File: Python/696. Count Binary Substrings.py
class Solution:
    def countBinarySubstrings(self, s):
        """
        :type s: str
        :rtype: int
        """
        last, now, ans = 0, 0, 0
        for i in range(len(s)):
            if s[i] == s[i-1]:
                now += 1
            else:
                last = now
                now = 1
            if last >= now:
                ans += 1
        return ans

File: C++/696. Count Binary Substrings.cpp
class Solution {
public:
    int countBinarySubstrings(string s) {
        int len = s.length();
        int last = 0,now = 1,ans = 0;
        for(int i=1;i<len;i++)
        {
            if(s[i]==s[i-1])
                now++;
            else
            {
                last = now;
                now = 1;
            }
            if(last>=now)
                ans++;
        }
        return ans;
        
    }
};